src/sdk/namespace_fs.js

```diff
@@ -682,7 +682,50 @@ class NamespaceFS {
         const fs_account_config = this.set_cur_fs_account_config(object_sdk);
         await this._load_bucket(params, fs_account_config);
 
+        const can_use_temp_file = Boolean(fs_account_config.backend === 'GPFS' && nb_native().fs.gpfs);
+        if (can_use_temp_file) {
+          const file_path = this._get_file_path(params);
+          const open_mode = 'wt';
+          // Uploading to a temp file then we will rename it.
+          let fs_xattr = to_fs_xattr(params.xattr);
+          // dbg.log0('NamespaceFS.upload_object:', this.upload_path, '->', file_path);
+          try {
+              if (params.copy_source) {
+                  const upload_id = uuidv4();
+                  const upload_path = path.join(this.bucket_path, this.get_bucket_tmpdir(), 'uploads', upload_id);
+                  const source_file_path = path.join(this.bucket_path, params.copy_source.key);
+                  await this._check_path_in_bucket_boundaries(fs_account_config, file_path);
+                  await this._make_path_dirs(upload_path, fs_account_config);
+                  await this._check_path_in_bucket_boundaries(fs_account_config, source_file_path);
+                  await this._fail_if_archived_or_sparse_file(fs_account_config, source_file_path);
+                  try {
+                      const same_inode = await this._is_same_inode(fs_account_config, source_file_path, file_path, fs_xattr);
+                      if (same_inode) return same_inode;
+                      // Doing a hard link.
+                      await nb_native().fs.link(fs_account_config, source_file_path, upload_path);
+                  } catch (e) {
+                      dbg.warn('NamespaceFS: COPY using link failed with:', e);
+                      fs_xattr = await this._copy_stream(source_file_path, upload_path, fs_account_config, fs_xattr);
+                  }
+              } else {
+                  // TODO: Take up only as much as we need (requires fine-tune of the semaphore inside the _upload_stream)
+                  // Currently we are taking config.NSFS_BUF_SIZE for any sized upload (1KB upload will take a full buffer from semaphore)
+                  fs_xattr = await buffers_pool_sem.surround_count(
+                      config.NSFS_BUF_SIZE,
+                      async () => this._upload_stream(params, this.bucket_path, file_path, open_mode, fs_account_config,
+                                                      object_sdk.rpc_client, fs_xattr)
+                  );
+              }
+              // TODO use file xattr to store md5_b64 xattr, etc.
+              const stat = await nb_native().fs.stat(fs_account_config, file_path);
+              return { etag: this._get_etag(stat, fs_xattr) };
+          } catch (err) {
+              this.run_update_issues_report(object_sdk, err);
+             throw this._translate_object_error_codes(err);
+          }
```

Comment:
It seems to me like pulling out the try-catch from being copied in these two cases should be a simple way to dedup a bit.

Reply:
Also I see that you introduced a "unique" 2 spaces indentation there. Can you please fix it to 4 spaces across this code?

Reply:
done